### ISSUE TITLE
Afterburners should be a burst of speed instead of flat engine power.

### ISSUE THREAD

**Issue by TimCabbage** [endless-sky/endless-sky#6898]
### Problem Description

Afterburners are underpowered (or may become overpowered after the proposed changes) but they're not different than engines except for fuel usage. I think those two overlap too much.

### Related Issue Links

https://discord.com/channels/251118043411775489/308902312741568512/981629711883042846

### Desired Solution

I think the solution would be to have afterburners be identical in function to normal engines (no fuel usage, just heat generation and power usage) except they should be working in burst.

Add variables: 
 > Heat capacity: How much heat this module can store before shutting down.
 > Heat threshold: How much heat this module can store before disabling it (stopping the afterburner button press) causes it to go into cooldown mode
 > Module Heat Generation: How much heat is generated **and not absorbed by the ship heat** but that lands in the heat capacity
 > Module Heat

Example timeline:
Heat Capacity=1000 
Heat Threshold=700, 
Heat Generation=250, 
FiringHeat=150

T=0
Afterburner is off, Module Heat = 300 
**Every second the afterburner is off, but module heat > 0, generate heat equal to FiringHeat, subtract FiringHeat from ModuleHeat**
Generate 150 heat

T=1
Afterburner is off, Module Heat = 150
Generate 150 heat

T=2 
Afterburner is off, Module Heat = 0
Generate 0 heat

T=3
Afterburner is on, Module Heat = 0;
Generate 150 heat
Add 100 extra heat to Module
Apply extra thrust

T=4
Afterburner is on, Module Heat = 100;
Generate 150 heat
Add 100 extra heat to Module
Apply extra thrust

(...)
T=8
Afterburner is off, Module Heat = 600;
Generate 150 heat

Because we didn't reach threshold, the module can still be enabled at any time.

T=9 
Afterburner is **ON**, Module Heat = 450;
Generate 150 heat

T=10
Afterburner is on, Module Heat = 550;
Generate 150 heat
Add 100 extra heat to Module
Apply extra thrust

T=11
Afterburner is on, Module Heat = 650;
Generate 150 heat
Add 100 extra heat to Module
Apply extra thrust

T=12
Afterburner is on, Module Heat = 750;
Generate 150 heat
Add 100 extra heat to Module
Apply extra thrust

T=13
Afterburner is on, Module Heat = 850;
Generate 150 heat
Add 100 extra heat to Module
Apply extra thrust

T=14
Afterburner is **off**, Module Heat = 850;
Generate 150 heat

Because we are above threshold, we can't reenable the afterburner until module heat falls below 700

The module will generate heat even if off until module heat reaches 0.

### Alternative Approaches

May still use fuel, that's a parallel decision to make.

**Comment by Zitchas** [comment]
Out of curiosity, have you tried using the Bellows-class afterburners? Those are constrained via fuel usage to be "jump" style afterburners instead of cruising afterburners. As in they provide a very heavy kick of acceleration, but they burn fuel fast that they discourage using them for more than just a short duration.

**Comment by TimCabbage** [comment]
Anything that uses fuel is out in my book because I really dislike being stranded in a system I just captured a valuable ship in :)
But I did use various afterburners at some point and didn't really like then in a cost vs benefit calculation.

**Comment by ravenshining** [comment]
If something like this is implemented, I think we should keep the existing afterburners in some capacity.  I really love setting up afterburner-only ships with ramscoops

**Comment by Zitchas** [comment]
I agree with Ravenshining: Differentiating afterburners so that there is a larger variety of attributes is good; but wholesale changing all of them to a different paradigm is probably not likely to happen. 

I'm sure you probably also realize the irony that removing the fuel requirement to make them strictly energy and heat restricted actually makes them *more* like engines in some ways...

That being said, the mechanic and idea is an interesting one, and it would be interesting to have a few options along these lines. That being said, this would require a lot of "outfit functioning as independent entity" code, which we don't have right now, or perhaps implementing some kind of heat lag capacity & release values to handle the production of heat that isn't counted as part of the ship's immediately, but is released over time. (although we could probably simulate this effect by having the afterburners a burn damage per thrust attribute, burn damage being heat damage over time, which is basically heat that will warm up the ship slowly over a period of time instead of instantly.
